rust/src/clib/test/nmstate_gen_diff_test.c

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include <assert.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <nmstate.h>
+
+int main(void) {
+	int rc = EXIT_SUCCESS;
+	const char *des_state = "---\n"
+		"interfaces:\n"
+		"  - name: eth1\n"
+		"    type: ethernet\n"
+		"    state: up\n"
+		"    ipv4:\n"
+		"      enabled: true\n"
+		"      dhcp: true\n"
+		"    ipv6:\n"
+		"      enabled: false";
+	const char *cur_state = "---\n"
+		"interfaces:\n"
+		"  - name: eth1\n"
+		"    type: ethernet\n"
+		"    state: up\n"
+		"    ipv4:\n"
+		"      enabled: true\n"
+		"      dhcp: true\n"
+		"    ipv6:\n"
+		"      enabled: true\n"
+		"      dhcp: true\n"
+		"      autoconf: true";
+	char *diff_state = NULL;
+	char *err_kind = NULL;
+	char *err_msg = NULL;
+
+	if (nmstate_generate_differences(des_state,
+					 cur_state,
+					 &diff_state,
+					 &err_kind,
+					 &err_msg) == NMSTATE_PASS) {
+		printf("%s\n", diff_state);
```

Comment:
How do you know that the `diff_state` is correct ? Can we add the string comparison here  ?

Reply:
This test is just make sure we do not have memory leak in c binding. Integration test case will check whether it works or not.

The yaml output might change indent or other format, so char compare in C is not reliable.

Reply:
Got it.